File: java/0150-evaluate-reverse-polish-notation/Solution.java
import java.util.Stack;

public class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();

        int second;

        for (String token : tokens) {
            if (token.equals("+")) {
                stack.push(stack.pop() + stack.pop());
            } else if (token.equals("-")) {
                second = stack.pop();
                stack.push(stack.pop() - second);
            } else if (token.equals("*")) {
                stack.push(stack.pop() * stack.pop());
            } else if (token.equals("/")) {
                second = stack.pop();
                stack.push(stack.pop() / second);
            } else {
                stack.push(Integer.parseInt(token));
            }
        }

        return stack.pop();
    }
}

File: c++/0150-evaluate-reverse-polish-notation/solution.cpp
#include <stack>
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> s;

        int first, second;

        for (int i = 0; i < tokens.size(); ++i) {
            if (tokens[i] == "+") {
                second = s.top();
                s.pop();

                first = s.top();
                s.pop();

                s.push(first + second);
            } else if (tokens[i] == "-") {
                second = s.top();
                s.pop();

                first = s.top();
                s.pop();

                s.push(first - second);
            } else if (tokens[i] == "*") {
                second = s.top();
                s.pop();

                first = s.top();
                s.pop();

                s.push(first * second);
            } else if (tokens[i] == "/") {
                second = s.top();
                s.pop();

                first = s.top();
                s.pop();

                s.push(first / second);
            } else {
                s.push(stoi(tokens[i]));
            }
        }

        return s.top();
    }
};
